packages/starlight/utils/route-data.ts

```diff
@@ -82,11 +81,10 @@ function getLastUpdated({ entry }: PageProps): Date | undefined {
 	const { lastUpdated: configLastUpdated } = config;
 
 	if (frontmatterLastUpdated ?? configLastUpdated) {
-		const currentFilePath = fileURLToPath(new URL('src/content/docs/' + entry.id, project.root));
```

Comment:
As you mentioned, considering you also fixed this for users with a custom `srcDir`, I think we should add a second changeset file for this change.